### PR TITLE
Feature/selective update

### PR DESCRIPTION
**Pull request by lyricnz**
Add the ability to update an explicit subset of the data available from the ESS switch, rather than all possible data (~7).

As mentioned in https://github.com/firstof9/tplink-ess-lib/issues/41#issuecomment-1526880111

### CODE REVIEW

**Comment by lyricnz** [comment]
Might need a version bump

File: tests/test_init.py [hunk 1]
```diff
@@ -264,6 +264,52 @@ async def test_update_data():
...
 
 
+async def test_partial_update_data():
+    """Test update data function with subset."""
+    with patch("tplink_ess_lib.network.socket.socket") as mock_socket:
+        mock_socket = mock_socket.return_value
+        packet1 = bytes.fromhex(base64.b64decode(TEST_PACKETS[9]).decode("utf-8"))
+        packet2 = bytes.fromhex(base64.b64decode(TEST_PACKETS[10]).decode("utf-8"))
+        packet3 = bytes.fromhex(base64.b64decode(TEST_PACKETS[2]).decode("utf-8"))
+        packet5 = bytes.fromhex(base64.b64decode(TEST_PACKETS[4]).decode("utf-8"))
```
Comment (lyricnz) [inline tests/test_init.py:274]: This code is copied from test_update_data() above.  Maybe the two tests could extract this prep into some constants for reuse....

Comment (firstof9) [reply tests/test_init.py:274]: Ya this was a quick cobbling together too

**Review by firstof9 — APPROVED** [review]
LGTM

File: tests/test_init.py (hunk 1 above)
Comment (lyricnz) [reply tests/test_init.py:274]: I'll fix this in https://github.com/firstof9/tplink-ess-lib/issues/43

**Pull request merged**